Added several new bindings

neslib opened the pull request:
- Small bugfix in igSliderFloat2
- Added ImDrawData_ScaleClipRects
- Added ImGuiTextFilter APIs
- Added ImGuiTextEditCallbackData
- Added ImGuiStorate APIs
- Added ImFontAtlas_GetGlyphRanges\* APIs

procedural commented:
I guess `cimgui.VC.db` slipped in accidentally? :)

neslib commented:
Oops. You can ignore that one.
Sorry about that. Should add that one to my global gitignore...

procedural commented:
You forgot `struct` keyword for `drawData` here: https://github.com/Extrawurst/cimgui/pull/18/files#diff-9dac42b77ccda07e6c9f40fb19c3c046R432
And `struct ImGuiTextFilter` isn't declared anywhere it seems, I get compiler warnings...

procedural commented on cimgui/cimgui.h:
```diff
@@ -417,13 +417,19 @@ CIMGUI_API struct ImFont*   ImFontAtlas_AddFontFromMemoryCompressedTTF(struct Im
 
 //ImDrawData
 CIMGUI_API void                 ImDrawData_DeIndexAllBuffers(struct ImDrawData* drawData);
+CIMGUI_API void					ImDrawData_ScaleClipRects(ImDrawData* drawData, struct ImVec2 sc);
```
`ImDrawData* drawData` should be `struct ImDrawData* drawData`

neslib commented:
Sorry again. I only use the Visual Studio solution and it compiled fine with my changes.
I added a new commit that fixes these issues (hopefully ;) I also removed the VC.db file.

Extrawurst commented:
Thank you guys so much. did not look into it in detail yet, @procedural does it look good to you now ?

procedural commented:
@Extrawurst yep, it's good now

Extrawurst commented:
@neslib is this also updating the API to a recent version or is this adding new bindings in the status om imgui that cimgui is binding to (v 1.49) ?

neslib commented:
@Extrawurst these new bindings still apply to v 1.49. They compile against the imgui submodule included with cimgui.

Extrawurst commented:
ok great, thanks

Extrawurst commented:
thank you very much!
